tower-spawn-ready/src/future.rs

```diff
@@ -0,0 +1,78 @@
+use crate::error::Error;
+use futures::{Async, Future, Poll};
+use tokio_executor::TypedExecutor;
+use tokio_sync::oneshot;
+use tower_service::Service;
+use tower_util::Ready;
+
+pub struct BackgroundReady<T, Request>
+where
+    T: Service<Request>,
+    T::Error: Into<Error>,
+{
+    ready: Ready<T, Request>,
+    tx: Option<oneshot::Sender<Result<T, Error>>>,
+}
+
+/// This trait allows you to use either Tokio's threaded runtime's executor or
+/// the `current_thread` runtime's executor depending on if `T` is `Send` or
+/// `!Send`.
+pub trait BackgroundReadyExecutor<T, Request>: TypedExecutor<BackgroundReady<T, Request>>
+where
+    T: Service<Request>,
+    T::Error: Into<Error>,
+{
+}
+
+impl<T, Request, E: TypedExecutor<BackgroundReady<T, Request>>> BackgroundReadyExecutor<T, Request>
+    for E
```

Comment:
Nit: The way rustfmt chose to format this is a little weird IMO. Maybe moving the bounds on `E` into the `where` clause would shorten the first line, making it not get rustfmt-ed this way? Also, IMO, it's a little easier to grok when all the bounds are in the same place.